include/cpr/connection_pool.h

```diff
@@ -0,0 +1,21 @@
+#ifndef CPR_CONNECTION_POOL_H
+#define CPR_CONNECTION_POOL_H
+
+#include <curl/curl.h>
+#include <memory>
+#include <mutex>
+
+namespace cpr {
+class ConnectionPool {
+  public:
+    ConnectionPool();
+    ConnectionPool(const ConnectionPool&) = default;
+    ConnectionPool& operator=(const ConnectionPool&) = delete;
+    void SetupHandler(CURL* easy_handler) const;
+
+  private:
+    std::shared_ptr<std::mutex> connection_mutex_;
+    std::shared_ptr<CURLSH> curl_sh_;
+};
+} // namespace cpr
+#endif 
```

Comment:
Can I get a bit of inline code documentation here? I know not all the existing code is documented well, but I try to change this with all new additions e.g. https://github.com/libcpr/cpr/pull/1168